### PR TITLE
News-US NewsDetails-01 : Displaying article (without actions)

### PR DESCRIPTION
**Pull request by mohamedHedi1990**

### CODE REVIEW

File: extension/war/src/main/webapp/groovy/social/webui/activity/UINewsFullActivity.gtmpl [hunk 1]
```diff
@@ -0,0 +1,503 @@
...
+  import java.text.SimpleDateFormat;
+
+  import org.apache.commons.lang.StringEscapeUtils;
+  import org.apache.commons.lang.StringUtils
+  
+  import java.util.Date;
+
+  import static org.exoplatform.social.webui.activity.BaseUIActivity.TEMPLATE_PARAM_COMMENT;
+
+  def mediumDateFormat = new SimpleDateFormat("d MMM yyyy");
```
Comment (thomasdelhomenie) [inline extension/war/src/main/webapp/groovy/social/webui/activity/UINewsFullActivity.gtmpl]: It is recommended to use new Java 8 Date/Time API, so to replace SimpleDateFormat with DateTimeFormatter.

File: extension/war/src/main/webapp/groovy/social/webui/activity/UINewsStreamActivity.gtmpl [hunk 2]
```diff
@@ -0,0 +1,559 @@
...
+	<div class="activityAvatar avatarCircle">
+		<a  href="$ownerUri">
+		  <img alt="$ownerName" src="$ownerAvatar"/>
+		</a>
+	</div>
+  </div><!--end activityTimeLine-->
+  <div class="boxContainer" id="boxContainer">
+	<div id="ContextBox${activity.id}" class="uiBox contentBox">
+		<div id="ActivityContextBox${activity.id}">
+		  <div class="heading">
```
Comment (thomasdelhomenie) [inline extension/war/src/main/webapp/groovy/social/webui/activity/UINewsStreamActivity.gtmpl]: According to the design, we don't need the header part with displayed in full mode, so this part should probably be deleted.
Also we should check if the template contains only the needed information, not more.

File: extension/war/src/main/webapp/groovy/social/webui/activity/UINewsStreamActivity.gtmpl [hunk 3]
```diff
@@ -0,0 +1,559 @@
...
+		                <div class="owner">
+        					<a  href="$ownerUri">$ownerName</a>
+        				</div>
+					<%}%>
+				</div>
+				<div class="dataInfor">
+					<span class="dateTime"><i class="uiIconClock  uiIconLightGray"></i>&nbsp;<a id="EditActivityTime${activity.id}" rel="tooltip" data-placement="bottom" href="<%=uicomponent.getActivityPermalink(activity.id)%>" <% if(activity.postedTime != activity.updated.getTime() ){%> title="$postedTimeTooltip"> $activityUpdatedTime <%} else {%> >$activityPostedTime <%}%></a></span>
+
+					<%
+					if( activityDeletable ) {
```
Comment (thomasdelhomenie) [inline extension/war/src/main/webapp/groovy/social/webui/activity/UINewsStreamActivity.gtmpl]: The design does not contain the Delete action, it should be removed IMO.

File: extension/war/src/main/webapp/groovy/social/webui/activity/UINewsStreamActivity.gtmpl [hunk 4]
```diff
@@ -0,0 +1,559 @@
...
+					  <% } else { %>
+						<li>
+							<a onclick="$likeActivityAction" rel="tooltip" data-placement="bottom" title="<%=_ctx.appRes("UIActivity.msg.LikeActivity"); %>" id="LikeLink${activity.id}" href="javascript:void(0);">
+							  <i class="uiIconThumbUp uiIconLightGray"></i>&nbsp;$totalLikesNum
+							</a>
+						</li>
+					  <% }%>
+					</ul>
+			</div><!--end actionBar-->
+			<% if (identityLikesNum > 0 || uicomponent.isLiked()) {%>
```
Comment (thomasdelhomenie) [inline extension/war/src/main/webapp/groovy/social/webui/activity/UINewsStreamActivity.gtmpl]: The design does not contain likes and comments. Should we keep them in the detailed view ?

File: extension/war/src/main/webapp/groovy/social/webui/activity/UINewsStreamActivity.gtmpl (hunk 3 above)
Comment (mohamedHedi1990) [reply extension/war/src/main/webapp/groovy/social/webui/activity/UINewsStreamActivity.gtmpl]: I want to note here that the UINewsStreamActivity.gtmpl  is used to display the news in stream activities and not for the full view mode. this part of code was already deleted in full mode (see UINewsFullActivity.gtmpl).

File: extension/war/src/main/webapp/groovy/social/webui/activity/UINewsStreamActivity.gtmpl (hunk 2 above)
Comment (mohamedHedi1990) [reply extension/war/src/main/webapp/groovy/social/webui/activity/UINewsStreamActivity.gtmpl]: I want to note here that the UINewsStreamActivity.gtmpl  is used to display the news in stream activities and not for the full view mode. this part of code was already deleted in full mode (see UINewsFullActivity.gtmpl).

File: extension/war/src/main/webapp/groovy/social/webui/activity/UINewsStreamActivity.gtmpl (hunk 3 above)
Comment (thomasdelhomenie) [reply extension/war/src/main/webapp/groovy/social/webui/activity/UINewsStreamActivity.gtmpl]: Arghh, my bad, I thought it was the full mode template...

File: component/service/src/main/java/org/exoplatform/social/rest/impl/activity/ActivityRestResourcesV1.java [hunk 5]
```diff
@@ -208,7 +208,41 @@ public Response updateActivityById(@Context UriInfo uriInfo,
     
     return EntityBuilder.getResponse(activityInfo.getDataEntity(), uriInfo, RestUtils.getJsonMediaType(), Response.Status.OK);
   }
-  
+
+  @PUT
+  @Path("click/{id}/{targetField}")
```
Comment (thomasdelhomenie) [inline component/service/src/main/java/org/exoplatform/social/rest/impl/activity/ActivityRestResourcesV1.java]: This endpoint is specific to the News activity, soo it should not be in the Social REST API. We should instead create a dedicated class to expose this new endpoint.

Comment (thomasdelhomenie) [inline component/service/src/main/java/org/exoplatform/social/rest/impl/activity/ActivityRestResourcesV1.java]: I would change the endpoint to a POST with the path /activities/{id}/click, and pass the click type as the body:
```javascript
{
  "name": "readMore"
}
```
The logic is that what we want to store/retrieve is the number of clicks on an activity (so the REST resource is a click), and each click is an addition to the number of clicks, so it should be a POST.
WDYT ?

File: webapp/resources/src/main/webapp/javascript/eXo/social/webui/UIActivity.js [hunk 6]
```diff
@@ -1087,7 +1087,27 @@
         loading.hide();
         button.show();
       });
-    }
+    },
+
+    clickOnNews: function (activityId) {
+    	var env = eXo.social.portal;
+		var baseRestUrl = env.context + '/' + env.rest + '/v1/social/activities/click/'+ activityId + '/';
```
Comment (thomasdelhomenie) [inline webapp/resources/src/main/webapp/javascript/eXo/social/webui/UIActivity.js]: Indentation is not good.

File: webapp/resources/src/main/webapp/javascript/eXo/social/webui/UIActivity.js [hunk 7]
```diff
@@ -1087,7 +1087,27 @@
...
+    	$("#seeMoreId").click(function() {
```
Comment (thomasdelhomenie) [inline webapp/resources/src/main/webapp/javascript/eXo/social/webui/UIActivity.js]: The same click events will be bound to all seeMoreId elements, so to all activities, no ? Did you check that it works fine with several news activities in the activity stream ?

File: extension/war/src/main/webapp/groovy/social/webui/activity/UINewsActivity.gtmpl [hunk 8]
```diff
@@ -1,558 +1,10 @@
...
-</div><!--activityStream-->
-<% } else { %> <!-- activity deleted -->
-<% uiform.begin() %>
-<div class="activityStream deleted">$labelActivityHasBeenDeleted</div>
-<% uiform.end() %>
-<% } %>
\ No newline at end of file
+def isFullViewMode = (Utils.getActivityID() != null);
+if(isFullViewMode) {
+_ctx.include("war:/groovy/social/webui/activity/UINewsFullActivity.gtmpl");
```
[the remark this answers is not in the document]
Comment (mohamedHedi1990) [reply extension/war/src/main/webapp/groovy/social/webui/activity/UINewsActivity.gtmpl]: Fixed.

File: extension/war/src/main/webapp/groovy/social/webui/activity/UINewsActivity.gtmpl [hunk 9]
```diff
@@ -1,558 +1,10 @@
...
+} else {
+_ctx.include("war:/groovy/social/webui/activity/UINewsStreamActivity.gtmpl");
```
[the remark this answers is not in the document]
Comment (mohamedHedi1990) [reply extension/war/src/main/webapp/groovy/social/webui/activity/UINewsActivity.gtmpl]: Fixed.

File: extension/war/src/main/webapp/groovy/social/webui/activity/UINewsFullActivity.gtmpl (hunk 1 above)
Comment (mohamedHedi1990) [reply extension/war/src/main/webapp/groovy/social/webui/activity/UINewsFullActivity.gtmpl]: Fixed.

File: extension/war/src/main/webapp/groovy/social/webui/activity/UINewsStreamActivity.gtmpl (hunk 4 above)
Comment (mohamedHedi1990) [reply extension/war/src/main/webapp/groovy/social/webui/activity/UINewsStreamActivity.gtmpl]: Fixed.

File: component/service/src/main/java/org/exoplatform/social/rest/impl/activity/ActivityRestResourcesV1.java (hunk 5 above)
Comment (mohamedHedi1990) [reply component/service/src/main/java/org/exoplatform/social/rest/impl/activity/ActivityRestResourcesV1.java]: Fixed.

Comment (mohamedHedi1990) [reply component/service/src/main/java/org/exoplatform/social/rest/impl/activity/ActivityRestResourcesV1.java]: Fixed.

File: webapp/resources/src/main/webapp/javascript/eXo/social/webui/UIActivity.js (hunk 6 above)
Comment (mohamedHedi1990) [reply webapp/resources/src/main/webapp/javascript/eXo/social/webui/UIActivity.js]: Fixed.

File: webapp/resources/src/main/webapp/javascript/eXo/social/webui/UIActivity.js (hunk 7 above)
Comment (mohamedHedi1990) [reply webapp/resources/src/main/webapp/javascript/eXo/social/webui/UIActivity.js]: Fixed.

**Pull request merged**